### PR TITLE
Updated Makefile for device-modbus-go

### PR DESCRIPTION
**Pull request by jamesrgregg**
- include docker build instructions in makefile for device-modbus-go

Signed-off-by: James Gregg <james.r.gregg@intel.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,5 +31,11 @@ prepare:
...
 	glide update
 
-run:
-	cd bin && ./edgex-launch.sh
\ No newline at end of file
+docker: $(DOCKERS)
+
+docker_device_modbus_go:
+	docker build \
+		--label "git_sha=$(GIT_SHA)" \
```
Comment (mhall119) [inline Makefile]: It doesn't look like GIT_SHA is being defined anywhere

Comment (jamesrgregg) [reply Makefile]: I was missing a build argument in .PHONY.
I pushed a minor change.

Comment (jamesrgregg) [reply Makefile]: The GIT_SHA should come as a global variable from the git commit hash.

Comment (mhall119) [reply Makefile]: You'll still need to add a line like this one: https://github.com/edgexfoundry/device-random/blob/master/Makefile#L13

Comment (jamesrgregg) [reply Makefile]: thank you @mhall119 I have pushed that change now.

File: Dockerfile [hunk 2]
```diff
@@ -14,7 +14,12 @@ RUN echo http://nl.alpinelinux.org/alpine/v3.6/main > /etc/apk/repositories; \
 RUN apk update && apk add make && apk add bash
 RUN apk add curl && apk add git && apk add openssh
```
Comment (JPWKU) [inline Dockerfile]: We could remove your script from below and add `glide` to the `apk add` command here.

Comment (JPWKU) [reply Dockerfile]: Analogous to what happens here in config-seed Dockerfile https://github.com/edgexfoundry/edgex-go/blob/master/docker/Dockerfile.config-seed#L28-L34.
The issue you are currently hitting after you download the binary is that it's not on the PATH.
Installing via apk will solve this.

Comment (JPWKU) [reply Dockerfile]: @jamesrgregg disregard everything I just said above, my apologies.  I just realized you said there was a compilation issue and I see it now.

Comment (jamesrgregg) [reply Dockerfile]: I also reverted back to the previous build image.  The error is indicating that gcc isn't in the path.
I can perhaps try to add gcc via apk.

**Comment by jamesrgregg** [comment]
recheck

**Pull request merged**